src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs

```diff
@@ -400,54 +400,41 @@ public void Execute(DocumentsOperationContext context, Transaction indexWriteTra
                 {
                     using (var existingReferenceDocument = _database.DocumentsStorage.Get(context, referencesOfReduceOutput.Key))
                     {
-                        if (existingReferenceDocument == null)
+                        var uniqueIds = referencesOfReduceOutput.Value;
+
+                        if (existingReferenceDocument != null)
                         {
-                            var referenceDoc = new DynamicJsonValue
-                            {
-                                [nameof(OutputReduceToCollectionReference.ReduceOutputs)] = new DynamicJsonArray(referencesOfReduceOutput.Value),
-                                [Constants.Documents.Metadata.Key] = new DynamicJsonValue
-                                {
-                                    [Constants.Documents.Metadata.Collection] = $"{_outputReduceToCollection}/References"
-                                }
-                            };
-
-                            using (var referenceJson = context.ReadObject(referenceDoc, "reference-of-reduce-output", BlittableJsonDocumentBuilder.UsageMode.ToDisk))
+                            if (existingReferenceDocument.Data.TryGet(nameof(OutputReduceToCollectionReference.ReduceOutputs), out BlittableJsonReaderArray existingIds) == false)
+                                ThrowIdsPropertyNotFound(existingReferenceDocument.Id);
```

Comment:
```suggestion
                                ThrowReduceOutputsPropertyNotFound(existingReferenceDocument.Id);
```

Reply:
Renamed in https://github.com/ravendb/ravendb/pull/10055/commits/5902446451ad63e2740e270632e414237edef4e5